builtin/providers/aws/resource_aws_elasticsearch_domain.go

```diff
@@ -274,6 +275,22 @@ func resourceAwsElasticSearchDomainRead(d *schema.ResourceData, meta interface{}
 
 	d.Set("arn", *ds.ARN)
 
+	// get tags
+	type ListTagsRequest struct {
+	    ARN string
+	}
+
+	req := ListTagsRequest{
+		ARN:   *ds.ARN
+	}
+
+	resp, err := elasticsearch.ListTags(req)
+	if err != nil {
+		return err
+	}
+
+	d.Set("tags", resp)
```

Comment:
this likely won't work, as the Tags are a schema set and the results of `ListTags` will be a slice of `[]*elasticsearch.Tags`. We have to do a few steps here, see ELB as a resource:

- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_elb.go#L205

Which leads to a `tagsELB.go` file, which we'll need for `elasticsearch` as well. I can likely help there if you need, otherwise it should be pretty easy to just copy `tagsELB.go`